benchmarks/src/construction_benchmarks.cpp

```diff
@@ -0,0 +1,56 @@
+#include <cstdint>
+#include <string>
+
+#include <benchmark/benchmark.h>
+
+#include "./benchmark_helpers.h"
+
+template<typename T>
+static void benchmark_default_onto_vector(benchmark::State& state)
+{
+    for (auto _ : state) {
+        auto vec = std::vector<T>();
+        for (auto i = 0; i < state.range(0); i++) {
+            vec.emplace_back(T());
```

Comment:
Sonarlint says you should not create T and then use emplace, because that defeats the purpose of emplace